domains/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -341,139 +341,291 @@ pub mod pallet {
     }
 
     impl<T: Config> Pallet<T> {
-        fn process_initial_sync(initial_sync: InitialSyncOf<T>) -> DispatchResult {
+        fn process_initial_sync(initial_sync: InitialSyncOf) -> DispatchResult {
+            let sync_committee_root =
+                merkleization::hash_tree_root_sync_committee(&initial_sync.current_sync_committee)
+                    .map_err(|_| Error::<T>::SyncCommitteeHashTreeRootFailed)?;
             Self::verify_sync_committee(
-                initial_sync.current_sync_committee.clone(),
+                sync_committee_root.into(),
                 initial_sync.current_sync_committee_branch,
                 initial_sync.header.state_root,
-                config::CURRENT_SYNC_COMMITTEE_DEPTH,
-                config::CURRENT_SYNC_COMMITTEE_INDEX,
+                config::CurrentSyncCommitteeDepth::get(),
+                config::CurrentSyncCommitteeIndex::get(),
             )?;
 
-            let period = Self::compute_current_sync_period(initial_sync.header.slot);
+            let period = Self::compute_sync_committee_period(initial_sync.header.slot);
 
             let block_root: H256 =
                 merkleization::hash_tree_root_beacon_header(initial_sync.header.clone())
                     .map_err(|_| Error::<T>::HeaderHashTreeRootFailed)?
                     .into();
 
-            Self::store_sync_committee(period, initial_sync.current_sync_committee);
+            Self::store_sync_committee(period, initial_sync.current_sync_committee)?;
             Self::store_finalized_header(block_root, initial_sync.header);
             Self::store_validators_root(initial_sync.validators_root);
 
             Ok(())
         }
 
-        fn process_sync_committee_period_update(
-            update: SyncCommitteePeriodUpdateOf<T>,
+        fn is_next_sync_committee_known(finalized_period: u64) -> bool {
+            <SyncCommittees<T>>::get(finalized_period + 1).pubkeys.len() != 0
+        }
+
+        fn stored_next_sync_committee(finalized_period: u64) -> SyncCommitteeOf {
+            <SyncCommittees<T>>::get(finalized_period + 1)
+        }
+
+        fn stored_current_sync_committee(finalized_period: u64) -> SyncCommitteeOf {
+            <SyncCommittees<T>>::get(finalized_period)
+        }
+
+        fn is_sync_committee_update(update: &LightClientUpdateOf) -> bool {
+            update.next_sync_committee_branch.len() != 0
+        }
+
+        fn is_finality_update(update: &LightClientUpdateOf) -> bool {
+            update.finality_branch.len() != 0
+        }
+
+        fn validate_light_client_update(
```

Comment:
Can you explain what light client is here? This seems to be introduced in this commit

Reply:
https://github.com/ethereum/consensus-specs/blob/dev/specs/altair/light-client/sync-protocol.md#lightclientupdate